### PR TITLE
Image layers extension: initial implementation

### PR DESCRIPTION
**Pull request by feloy**

### CODE REVIEW

**Review by deboer-tim — APPROVED** [review]

File: src/extension.ts [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+  const history: History[] = config.history;
+  const layersResult: extensionApi.ImageFilesystemLayer[] = [];
+  for (const layer of layers) {
+    const currentLayer: extensionApi.ImageFilesystemLayer = {
+      id: layer.substring(0, 12),
+    };
+    const layerTar = path.join(tmpdir, layer);
+    await nodeTar.list({
+      file: layerTar,
+      onentry: (entry: ExtendedNodeEntry) => {
```
[the remark this answers is not in the document]
Comment (feloy) [reply src/extension.ts:89]: I think I introduced this type in the previous version as linkpath was not in the `ReadEntry` type. But I can see it now. Let's remove this `ExtendedNodeEntry` type and use the native one instead

**Review by deboer-tim — APPROVED** [review]
Just needs DCO fix.

**Review by benoitf — APPROVED** [review]
it's missing all CI jobs but I guess they'll come after

I see no issue on iterating on the code as it's not yet part of the catalog

File: tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    "outDir": "dist",
+    "skipLibCheck": true,
+    "types": [
+        "node",
+    ]
+},
+"include": [
+    "src",
+    "types/*.d.ts",
+    "../../types/**/*.d.ts"
```
Comment (benoitf) [inline tsconfig.json:22]: should be removed as it's part of podman desktop

File: vitest.config.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ ***********************************************************************/
+
+import path from 'node:path';
+
+const config = {
+  test: {
+  },
+  resolve: {
+    alias: {
+      '@podman-desktop/api': path.resolve(__dirname, '__mocks__/@podman-desktop/api.js'),
```
Comment (benoitf) [inline vitest.config.js:26]: this directory is not existing (to mock all the calls to `@podman-desktop/api` so it should be either removed there or we add the files

**Comment by benoitf** [comment]
I see no tests as well

File: package.json [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    "test": "vitest run --coverage",
+    "test:watch": "vitest watch --coverage",
+    "format:check": "prettier --check \"src/**/*.ts\"",
+    "format:fix": "prettier --write \"src/**/*.ts\"",
+    "lint:check": "eslint . --ext js,ts,tsx",
+    "lint:fix": "eslint . --fix --ext js,ts,tsx",
+    "watch": "vite build -w"
+  },
+  "dependencies": {
+    "@types/node": "^18",
```
Comment (benoitf) [inline package.json:24]: should be 20

File: package.json [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    "tar": "^7.4.0"
+  },
+  "devDependencies": {
+    "@podman-desktop/api": "^1.12.0-202407181817-86c7ee8",
```
Comment (benoitf) [inline package.json:28]: should be a dev dependency

File: tsconfig.json (hunk 2 above)
Comment (benoitf) [reply tsconfig.json:22]: I mean, the folder types and ../../ it's when you're inside Podman Desktop

**Review by benoitf — APPROVED** [review]
I hope some tests will come 🙏

**Pull request merged**